### PR TITLE
Mj static on cdn

### PR DESCRIPTION
**Pull request by jez4**
Workflow has been modified to put the files into staging and production folders. Based on the branch.
gsutil ... -Z ... should be doing the compression.
Templates are taking the path STATIC_URL.

I had an idea that environment of the docker container should tell the application in which environment it is running. Staging, Production, or Dev on local. This way we could assign variables based on environment. Or maybe you solved this somehow different.

### CODE REVIEW

File: .github/workflows/pythonapp.yml [hunk 1]
```diff
@@ -43,6 +44,48 @@ jobs:
...
+        uses: actions/checkout@master
+
+      - uses: GoogleCloudPlatform/github-actions/setup-gcloud@master
+        with:
+          version: '270.0.0'
+          service_account_email: ${{ secrets.SA_EMAIL }}
+          service_account_key: ${{ secrets.GOOGLE_APPLICATION_CREDENTIALS }}
+
+      - run: |
+             gcloud auth configure-docker
```
Comment (hnykda) [inline .github/workflows/pythonapp.yml:62]: Do you think this is needed? It's for the docker auth, that's useful for building/pushing to docker registry.

File: src/server/main.py [hunk 2]
```diff
@@ -23,25 +25,35 @@
 @app.get("/")
 async def model(request: Request) -> Response:
     """serve the main model visualization"""
-    return templates.TemplateResponse("model.html", {"request": request})
+    return templates.TemplateResponse(
+        "model.html", {"request": request, "STATIC_URL": STATIC_URL}
```
Comment (hnykda) [inline src/server/main.py:29]: What about having a function:

```
def get_context(request: Request, **kwargs):
  return {"request": request, "STATIC_URL": STATIC_URL, **kwargs}

...
return templates.TemplateResponse("model.html", get_context(request, message=message, ...))
```

so it doesn't violate Don't Repeat Yourself principle?

File: src/server/main.py [hunk 3]
```diff
@@ -9,6 +9,8 @@
 from server.config import CONFIG
 
 
+STATIC_URL = "https://storage.googleapis.com/static-covid/static/staging"
```
Comment (hnykda) [inline src/server/main.py:12]: I would initialize this from the env variable directly, so:

```
STATIC_URL = os.getenv("STATIC_URL", "/static")
```

Comment (hnykda) [reply src/server/main.py:12]: And then:
1. add to https://github.com/epidemics/covid/blob/master/deploy/chart/values.staging.yaml (similarly for values.production.yaml):
```
env:
  staticUrl: "https://storage.googleapis.com/static-covid/static/staging"
```
2. to https://github.com/epidemics/covid/blob/master/deploy/chart/templates/configmap.yaml#L6 add
```
  STATIC_URL: {{ .Values.env.staticUrl | quote }}
```

Comment (hnykda) [reply src/server/main.py:12]: I think that would all make it work out of the box with docker-compose/locally and would still fetch from gcs in deployments.

Also, please remove ENABLE_NOTION env, we don't use it for anything anymore.

File: .github/workflows/pythonapp.yml (hunk 1 above)
Comment (jez4) [reply .github/workflows/pythonapp.yml:62]: Frankly, do not know. Seemed important :)

File: src/server/main.py (hunk 2 above)
Comment (jez4) [reply src/server/main.py:29]: Yep, great idea.

File: src/server/main.py (hunk 3 above)
Comment (jez4) [reply src/server/main.py:12]: ok

File: .github/workflows/pythonapp.yml (hunk 1 above)
Comment (hnykda) [reply .github/workflows/pythonapp.yml:62]: I would guess that the service account key specification in above could be enough. Would take this out and see

File: src/server/main.py (hunk 3 above)
Comment (jez4) [reply src/server/main.py:12]: Yes, I could confirm that local docker is able to use GCS static files. Used it this way.

**Pull request merged**